src/test/java/org/opensearch/knn/integ/ModeAndCompressionIT.java

```diff
@@ -50,7 +52,7 @@ public class ModeAndCompressionIT extends KNNRestTestCase {
 
     private static final int DIMENSION = 16;
     private static final int NUM_DOCS = 20;
-    private static final int K = 2;
+    private static final int K = NUM_DOCS;
```

Comment:
we need to use K=NUM_DOCS here because there can be equal scores in the firstPass of search hence the results are not consistent in runs. With this atleast we ensure that all docs are searched and their scores are matched.